phoenix-core/src/main/java/org/apache/hadoop/hbase/regionserver/IndexHalfStoreFileReaderGenerator.java

```diff
@@ -395,7 +395,10 @@ public KeyValueScanner preStoreScannerOpen(final ObserverContext<RegionCoprocess
                         .getReader().getHFileReader().hasMVCCInfo(), readPt));
             }
         }
-        keyValueScanners.addAll(((HStore) store).memstore.getScanners(readPt));
+        List<KeyValueScanner> memstoreScanners = ((HStore) store).memstore.getScanners(readPt);
+        if(memstoreScanners != null) {
```

Comment:
What is the case that this would ever be null?

Reply:
This is the line where it might return null 
https://github.com/apache/hbase/blob/branch-1.3/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/DefaultMemStore.java#L637